include/cuda/std/detail/libcxx/include/optional

```diff
@@ -700,76 +763,84 @@ public:
     constexpr explicit optional(_InPlaceT, _Args&&... __args)
         : __base(in_place, _CUDA_VSTD::forward<_Args>(__args)...) {}
 
-    template <class _Up, class... _Args, class = _EnableIf<
-        is_constructible_v<value_type, initializer_list<_Up>&, _Args...>>
+    template <class _Up, class... _Args, class = __enable_if_t<
+        _LIBCUDACXX_TRAIT(is_constructible, value_type, initializer_list<_Up>&, _Args...)>
     >
     _LIBCUDACXX_INLINE_VISIBILITY
     constexpr explicit optional(in_place_t, initializer_list<_Up> __il, _Args&&... __args)
         : __base(in_place, __il, _CUDA_VSTD::forward<_Args>(__args)...) {}
 
-    template <class _Up = value_type, _EnableIf<
-        _CheckOptionalArgsCtor<_Up>::template __enable_implicit<_Up>()
+    template <class _Up = value_type, __enable_if_t<
+        _CheckOptionalArgsCtor<_Up>::template __enable_implicit<_Up>::__value
```

Comment:
I believe we should remove all these complicated indirections and turn them into static constexpr variables.

But we can do that in a follow-up PR